### PR TITLE
Add new runC test

### PR DESCRIPTION
**Pull request by drpaneas**

### CODE REVIEW

**Comment by okurz** [comment]
+1

**Comment by drpaneas** [comment]
Test run verfication:

- SLES12SP3 [mau-extratests](http://ultron.suse.de/tests/1132#step/runc/111)
- Tumbleweed [extratest_in_textmode](http://ultron.suse.de/tests/1133#step/runc/111)
- MicroOS [vmx](http://ultron.suse.de/tests/1131#step/runc/111)

File: tests/console/runc.pm [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    select_console("root-console");
+
+    # runC cannot create or extract the root filesystem on its own. Use Docker to create it.
+    record_info 'Setup', 'Setup the environment';
+    setup;
+    create_oci_bundle;
+
+    # installation of runc package
+    record_info 'Test #1', 'Test: Installation';
+    if (is_caasp && check_var('FLAVOR', 'DVD') && !check_var('SYSTEM_ROLE', 'plain')) {
```
Comment (kravciak) [inline tests/console/runc.pm]: You should be able to use is_caasp('DVD') - you don't have to check flavor separately

Comment (drpaneas) [reply tests/console/runc.pm]: Ah, yes! That's even more *human-readable* ;)

So, if I understood correctly, I have to replace:
`if (is_caasp && check_var('FLAVOR', 'DVD') && !check_var('SYSTEM_ROLE', 'plain')) {`
with:
`if (is_caasp('DVD') && !check_var('SYSTEM_ROLE', 'plain')) { `

is this correct?

**Comment by Tracerneo** [comment]
+1

File: tests/console/runc.pm (hunk 1 above)
Comment (Tracerneo) [reply tests/console/runc.pm]: Correct.
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/220e44a63d410ef6ef35dda95ea649499fb0dbbf/lib/utils.pm#L273-L287

File: tests/console/runc.pm [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    assert_script_run('ls -l config.json');
+    script_run('cp config.json config.json.backup');
+
+    # Modify the configuration to run the container in background
+    assert_script_run("sed -i -e '/\"terminal\":/ s/: .*/: false,/' config.json");
+    assert_script_run("sed -i -e 's/\"sh\"/\"echo\", \"Kalimera\"/' config.json");
+
+    # Run (create, start, and delete) the container after it exits
+    record_info 'Test #3', 'Test: Use the run command';
+    validate_script_output('runc run test1', sub { m/Kalimera/ });
```
Comment (kravciak) [inline tests/console/runc.pm]: It's up to your preference, but using grep is faster and maybe more readable - I think validate_script_output was meant for things like https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/puppet.pm#L34

I would use this instead:
assert_script_run("runc run test1 | grep Kalimera")

Comment (drpaneas) [reply tests/console/runc.pm]: good point ;) Let me try that

**Comment by drpaneas** [comment]
new verification: MicroOS [vmx](http://ultron.suse.de/tests/1139#step/runc/69)

**Pull request merged**